core/src/main/scala/za/co/absa/fadb/DBEngine.scala

```diff
@@ -16,57 +16,69 @@
 
 package za.co.absa.fadb
 
-import scala.concurrent.{ExecutionContext, Future}
+import cats.Monad
+import cats.implicits.toFunctorOps
+import za.co.absa.fadb.exceptions.StatusException
+
 import scala.language.higherKinds
 
 /**
-  * A basis to represent a database executor
-  */
-trait DBEngine {
+ *  `DBEngine` is an abstract class that represents a database engine.
+ *  It provides methods to execute queries and fetch results from a database.
+ *  @tparam F - The type of the context in which the database queries are executed.
+ */
+abstract class DBEngine[F[_]: Monad] {
 
   /**
-    * A type representing the (SQL) query within the engine
-    * @tparam T - the return type of the query
-    */
-  type QueryType[T] <: Query[T]
+   *  A type representing the (SQL) query within the engine
+   *  @tparam R - the return type of the query
+   */
+  type QueryType[R] <: Query[R]
+  type QueryWithStatusType[R] <: QueryWithStatus[_, _, R]
 
-  implicit val executor: ExecutionContext
+  /**
+   *  The actual query executioner of the queries of the engine
+   *  @param query  - the query to execute
+   *  @tparam R     - return the of the query
+   *  @return       - sequence of the results of database query
+   */
+  protected def run[R](query: QueryType[R]): F[Seq[R]]
 
   /**
-    * The actual query executioner of the queries of the engine
-    * @param query  - the query to execute
-    * @tparam R     - return the of the query
-    * @return       - sequence of the results of database query
-    */
-  protected def run[R](query: QueryType[R]): Future[Seq[R]]
+   *  The actual query executioner of the queries of the engine with status
+   *  @param query  - the query to execute
+   *  @tparam R     - return the of the query
+   *  @return       - sequence of the results of database query
```

Comment:
So should it return a sequence or not? I believe now it returns a single result.

Reply:
good catch, it returns single result